terraform/modules/eks/iam.tf

```diff
@@ -14,6 +13,47 @@ resource "aws_iam_role" "node-role" {
     Version = "2012-10-17"
   })
 
+  inline_policy {
+    name = "aws-efs-csi-driver-policy"
+
+    policy = jsonencode({
```

Comment:
dumping entire policy document as JSON could get messy. You can externalize it for better read and responsibility segregation.

```
data "aws_iam_policy_document" "efs-csi-driver-policy" {
  statement {
    actions = [
      "elasticfilesystem:DescribeAccessPoints",
      "elasticfilesystem:DescribeFileSystems",
      "elasticfilesystem:DescribeMountTargets",
    ]

    resources = [
      var.efs_file_system_arn
    ]
  }

  statement {
    actions = [
      "elasticfilesystem:CreateAccessPoint"
    ]

    resources = [
      var.efs_file_system_arn,
    ]

    condition {
      test     = "StringLike"
      variable = "aws:RequestTag/efs.csi.aws.com/cluster"
      values = ["true"]
    }
  }

  statement {
    actions = [
      "elasticfilesystem:DeleteAccessPoint"
    ]

    resources = [
      var.efs_file_system_arn,
    ]

    condition {
      test     = "StringEquals"
      variable = "aws:ResourceTag/efs.csi.aws.com/cluster"
      values = ["true"]
    }
  }
}
```
and then consume it as data json in `aws_iam_role.inline_policy`

```
inline_policy {
  name   = "efs-csi-driver-policy"
  policy = data.aws_iam_policy_document.efs-csi-driver-policy.json
}
```